101-azure-database-migration-service/azuredeploy.json

```diff
@@ -137,81 +112,193 @@
+                        "publisher": "Microsoft.Compute",
+                        "type": "CustomScriptExtension",
+                        "typeHandlerVersion": "1.9",
+                        "autoUpgradeMinorVersion": true,
+                        "settings": {
+                            "fileUris": "[variables('scriptFiles')]"
+                        },
+                        "protectedSettings": {
+                            "commandToExecute": "[concat('powershell -ExecutionPolicy Unrestricted -File ', './', variables('scriptLocation'), ' ', variables('scriptParameters'))]"
+                        }
+                    }
+                }
+            ],
             "dependsOn": [
-                "[resourceId('Microsoft.Network/networkInterfaces', variables('sourceNicName'))]",
-                "[resourceId('Microsoft.Storage/storageAccounts', variables('storageAccountName'))]"
+                "[if(parameters('createPublicIP'), resourceId('Microsoft.Network/networkInterfaces', concat(variables('sourceNicName'), 'public')), resourceId('Microsoft.Network/networkInterfaces', concat(variables('sourceNicName'), 'private')))]",
+                "[if(parameters('createPublicIP'), resourceId('Microsoft.Network/networkInterfaces', concat(variables('sourceNicName'), 'public')), resourceId('Microsoft.Network/networkInterfaces', concat(variables('sourceNicName'), 'private')))]"
             ]
         },
         {
             "type": "Microsoft.DataMigration/services",
-            "sku":
-            {
+            "sku": {
                 "name": "GeneralPurpose_4vCores",
                 "tier": "General Purpose",
                 "size": "4 vCores"
             },
             "name": "[variables('DMSServiceName')]",
             "apiVersion": "2017-11-15-privatepreview",
             "location": "[parameters('location')]",
-            "properties":
-            {
+            "properties": {
                 "virtualSubnetId": "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('adVNet'), variables('defaultSubnetName'))]"
             },
+            "resources": [
+                {
+                    "type": "projects",
+                    "name": "SqlToSqlDbMigrationProject",
+                    "apiVersion": "2017-11-15-privatepreview",
+                    "location": "[parameters('location')]",
+                    "properties": {
+                        "sourcePlatform": "SQL",
+                        "targetPlatform": "SQLDB"
+                    },
+                    "dependsOn": [
+                        "[resourceId('Microsoft.DataMigration/services', variables('DMSServiceName'))]"
+                    ]
+                }
+            ],
             "dependsOn": [
                 "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('adVNet'), variables('defaultSubnetName'))]"
             ]
         },
         {
+            "condition": "[parameters('createPublicIP')]",
```

Comment:
this needs to be combined into a single resource...  see: https://github.com/Azure/azure-quickstart-templates/blob/master/100-marketplace-sample/azuredeploy.json#L273 for how to handle the conditional publicIP

Reply:
I tried using "id": "[if(parameters('createPublicIP'), resourceId('Microsoft.Network/publicIPAddresses/', variables('publicIPSourceServer')), json('null'))]"  but I get an error saying

{
  "error": {
    "code": "InvalidRequestFormat",
    "message": "Cannot parse the request.",
    "details": [
      {
        "code": "MissingJsonReferenceId",
        "message": "Value for reference id is missing. Path properties.ipConfigurations[0].properties.publicIPAddress."
      }
    ]
  }
}

when I select false for createpublicIp paramater